tests/nmap_agent_test.py

```diff
@@ -810,3 +810,176 @@ def testAgent_whenHostHaveOs_fingerprintMessageShouldHaveOs(
+                                        "@osgen": "10",
+                                        "@accuracy": "88",
+                                        "cpe": "cpe:/o:microsoft:windows_10:1511",
+                                    }
+                                ],
+                            }
+                        ]
+                    ]
+                },
+            }
+        }
+    }
+
+    mocker.patch(
+        "agent.nmap_wrapper.NmapWrapper.scan_hosts",
+        return_value=(product_fake_output, ""),
+    )
+
+    nmap_test_agent.process(ipv4_msg)
+
+    assert len(agent_mock) == 4
+    assert agent_mock[0].selector == "v3.asset.ip.v4.port.service"
+    assert agent_mock[1].selector == "v3.report.vulnerability"
+    fingerprint_msg = agent_mock[2]
+    assert fingerprint_msg.selector == "v3.fingerprint.ip.v4.service.library"
+    assert fingerprint_msg.data["host"] == "127.0.0.1"
+    assert fingerprint_msg.data["library_name"] == "Windows"
+    assert fingerprint_msg.data["library_version"] == "10"
+
+
+def testAgent_whenOsMatchIsEmptyList_fingerprintMessageShouldHaveOs(
+    nmap_test_agent: nmap_agent.NmapAgent,
+    agent_mock: List[message.Message],
+    ipv4_msg: message.Message,
+    agent_persist_mock: Dict[Union[str, bytes], Union[str, bytes]],
+    mocker: plugin.MockerFixture,
+) -> None:
+    """Ensure the agents emits the detected library name with its version."""
+    del agent_persist_mock
+    product_fake_output = {
+        "nmaprun": {
+            "host": {
+                "address": {"@addr": "127.0.0.1", "@addrtype": "ipv4"},
+                "ports": {
+                    "port": {
+                        "@portid": "22",
+                        "@protocol": "tcp",
+                        "state": {
+                            "@state": "open",
+                            "@reason": "syn-ack",
+                            "@reason_ttl": "0",
+                        },
+                        "service": {
+                            "@name": "ssh",
+                            "@product": "OpenSSH",
+                            "@version": "7.4",
+                            "cpe": "cpe:/a:openbsd:openssh:7.4",
+                        },
+                    }
+                },
+                "os": {"osmatch": []},
+            }
+        }
+    }
+
+    mocker.patch(
+        "agent.nmap_wrapper.NmapWrapper.scan_hosts",
+        return_value=(product_fake_output, ""),
+    )
+
+    nmap_test_agent.process(ipv4_msg)
+
+    assert len(agent_mock) == 2
+
+
+def testAgent_whenOsClassIsEmptyList_fingerprintMessageShouldHaveOs(
+    nmap_test_agent: nmap_agent.NmapAgent,
+    agent_mock: List[message.Message],
+    ipv4_msg: message.Message,
+    agent_persist_mock: Dict[Union[str, bytes], Union[str, bytes]],
+    mocker: plugin.MockerFixture,
+) -> None:
+    """Ensure the agents emits the detected library name with its version."""
```

Comment:
It doesn't look like this is what the method does

Reply:
done